# HG changeset patch
# User sgehwolf
# Date 1450425347 -3600
#      Fri Dec 18 08:55:47 2015 +0100
# Node ID 9a57d01ddf0309b4250ad6767dd4cc26b5c8626f
# Parent  5f2728295878fc80d8d14b4304abe9629e6893cd
6425769, PR2859: Allow specifying an address to bind JMX remote connector
Reviewed-by: jbachorik, dfuchs

diff -r 5f2728295878 -r 9a57d01ddf03 src/share/vm/services/diagnosticCommand.cpp
--- openjdk/hotspot/src/share/vm/services/diagnosticCommand.cpp	Fri Apr 01 04:06:33 2016 +0100
+++ openjdk/hotspot/src/share/vm/services/diagnosticCommand.cpp	Fri Dec 18 08:55:47 2015 +0100
@@ -437,6 +437,10 @@
   ("config.file",
    "set com.sun.management.config.file", "STRING", false),
 
+  _jmxremote_host
+  ("jmxremote.host",
+   "set com.sun.management.jmxremote.host", "STRING", false),
+
   _jmxremote_port
   ("jmxremote.port",
    "set com.sun.management.jmxremote.port", "STRING", false),
@@ -516,6 +520,7 @@
 
   {
     _dcmdparser.add_dcmd_option(&_config_file);
+    _dcmdparser.add_dcmd_option(&_jmxremote_host);
     _dcmdparser.add_dcmd_option(&_jmxremote_port);
     _dcmdparser.add_dcmd_option(&_jmxremote_rmi_port);
     _dcmdparser.add_dcmd_option(&_jmxremote_ssl);
@@ -586,6 +591,7 @@
     }
 
     PUT_OPTION(_config_file);
+    PUT_OPTION(_jmxremote_host);
     PUT_OPTION(_jmxremote_port);
     PUT_OPTION(_jmxremote_rmi_port);
     PUT_OPTION(_jmxremote_ssl);
diff -r 5f2728295878 -r 9a57d01ddf03 src/share/vm/services/diagnosticCommand.hpp
--- openjdk/hotspot/src/share/vm/services/diagnosticCommand.hpp	Fri Apr 01 04:06:33 2016 +0100
+++ openjdk/hotspot/src/share/vm/services/diagnosticCommand.hpp	Fri Dec 18 08:55:47 2015 +0100
@@ -280,6 +280,7 @@
   // com.sun.management is omitted
 
   DCmdArgument<char *> _config_file;
+  DCmdArgument<char *> _jmxremote_host;
   DCmdArgument<char *> _jmxremote_port;
   DCmdArgument<char *> _jmxremote_rmi_port;
   DCmdArgument<char *> _jmxremote_ssl;
